server/handlers/clients.js

```diff
@@ -190,14 +184,19 @@ function addConfigHandler(req, res) {
   const id = req.params.id;
 
   if (clients.get(id))
-    return res
-      .status(409)
-      .send({ message: `A client with the id '${id}' already exists` });
+    return res.status(409).send({
+      error: {
+        message: `A client with the id '${id}' already exists`,
+        code: 409
+      }
+    });
 
+  req.type = 'add';
   return updateOrAdd(req, res);
 }
 
 function updateConfigHandler(req, res) {
+  req.type = 'update';
```

Comment:
it also seems like you could do the `update` step in here and then pass along to a function that just `add`s

Reply:
Yep! I like that.